### PR TITLE
Added support for AD717x-AD411x family of devices

### PR DESCRIPTION
**Pull request by jansunil**
1) Added ad717x.py with supported parts and attributes
2) Added an example script for ad717x
3) Added a test script
4) Updated supported_parts.md
5) Updated __init__.py
6) Added adi.ad717x.rst

Signed-off-by: Janani Sunil <Janani.Sunil@analog.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A - Tested the ad717x_example.py with a 1.8v input at the analog inputs

**Test Configuration**:
* Hardware: EVAL-AD4111SDZ with the SDP-K1 board
* OS: Windows 10

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jansunil** [comment]
V2: Updated the spacing in docstrings

**Comment by jansunil** [comment]
V3 Changes: Updated index.rst with ad717x

**Comment by jansunil** [comment]
V4 Changes: Corrected spacing in code documentation

**Comment by jansunil** [comment]
V5 Changes: Added a period at the end of the code documentation

**Comment by jansunil** [comment]
V6 Changes: Added one line before and after class docstring, removed lines before and after function docstring

**Comment by jansunil** [comment]
V7 Changes: Corrected syntax of constructor docstring

**Comment by jansunil** [comment]
@tfcollins , Any review comments on this PR?

**Comment by tfcollins** [comment]
Sorry for the delay. I was out a few weeks

**Comment by jansunil** [comment]
Hi @tfcollins ,
Could you please help in understanding what the Lint test is all about? The details show that the 'black' step has failed, although it does not point to any line in the code.

**Comment by jansunil** [comment]
Hi @tfcollins , 
Any inputs on the above build errors?

**Comment by tfcollins** [comment]
Please run the precommit lint to fix the formatting issue

**Comment by jansunil** [comment]
@tfcollins ,
Thanks for the link to the pre-commit step. The changes post execution of the command have been fixed.

**Comment by jansunil** [comment]
@tfcollins ,

I do not see any specific comments within the code. Am I missing something ?

**Comment by tfcollins** [comment]
> @tfcollins ,
> 
> I do not see any specific comments within the code. Am I missing something ?

Look above and in the PR changed files: https://github.com/analogdevicesinc/pyadi-iio/pull/326/files

File: adi/__init__.py [hunk 1]
```diff
@@ -151,6 +151,8 @@
 
 from adi.adpd1080 import adpd1080
 
+from adi.ad717x import ad717x
+
```
Comment (pmallick-adi) [inline adi/__init__.py:155]: Does it support AD411x devices as well?

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/__init__.py (hunk 1 above)
Comment (jansunil) [reply adi/__init__.py:155]: Yes, it does.

File: supported_parts.md [hunk 2]
```diff
@@ -44,6 +44,7 @@
 - AD5696R
 - AD6676
 - AD7124
+- AD717x
```
[the remark this answers is not in the document]
Comment (jansunil) [reply supported_parts.md]: Added actual part names

File: adi/ad717x.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            "ad7177-2"
+        ]
+
+        self._ctrl = None
+
+        if not device_name:
+            device_name = compatible_parts[0]
+        else:
+            if device_name not in compatible_parts:
+                raise Exception("Not a compatible device: " + device_name)
```
[the remark this answers is not in the document]
Comment (jansunil) [reply adi/ad717x.py]: The string has been changed to an f-string

File: adi/ad717x.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+        # Select the device matching device_name as working device
+        for device in self._ctx.devices:
+            if device.name == device_name:
+                self._ctrl = device
+                self._rxadc = device
+                break
+
```
[the remark this answers is not in the document]
Comment (jansunil) [reply adi/ad717x.py:83]: Added a check for the mentioned variables

File: adi/ad717x.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        ret = None
+
+        if isinstance(val, np.int16):
+            ret = val * _scale
+
+        if isinstance(val, np.ndarray):
+            ret = [x * _scale for x in val]
+
+        return ret
```
[the remark this answers is not in the document]
Comment (jansunil) [reply adi/ad717x.py:143]: A 'None' check has been added now

File: examples/ad717x_example.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+ad717x_dev = adi.ad717x("ip:analog")
+
+chn = 0
+ad717x_dev._rx_data_type = np.int32
+ad717x_dev.rx_output_type = "SI"
+ad717x_dev.rx_enabled_channels = [chn]
+ad717x_dev.rx_buffer_size = 100
+
+ad717x_dev.channel[0].raw
```
[the remark this answers is not in the document]
Comment (jansunil) [reply examples/ad717x_example.py]: This line has been removed now

**Review by tfcollins — APPROVED** [review]

**Pull request merged**